skupper/network/roles/skupper_link/tasks/execute.yml

```diff
@@ -29,12 +29,22 @@
   register: tokentemp
 
 # Saving secret content into the temporary
-- name: Saving token file
+- name: Saving token file from hostvar
   ansible.builtin.copy:
     content: "{{ hostvars[item.host]['generatedToken'] }}"
     dest: "{{ tokentemp.path }}"
     mode: '0644'
     remote_src: true
+  when: hostvars[item.host]['generatedToken'] is defined
+
+# Saving secret content into the temporary
+- name: Saving token file from var
+  ansible.builtin.copy:
+    content: "{{ tokens[item['host']]['generatedToken'] }}"
+    dest: "{{ tokentemp.path }}"
+    mode: '0644'
+    remote_src: true
+  when: tokens[item['host']]['generatedToken'] is defined
```

Comment:
If the idea is to provide a static token, instead of using the one generated
by the skupper_token role, I think it would make more sense to name it as:
`hostvars[item.host]['staticToken']` instead of calling it generated tokens.<host>.generatedToken as well.

Then in the inventory file, you could just add a `staticToken` variable to the node providing such token.

Reply:
I might not need this after all, what I'm trying to do is this

Have a play book the Init and get token for a site store the token in vault.

Have another playbook that can link to the first playbook but retrieve the token first from vault, I think I can do what I need without the role being modified.  I will close this if so

Reply:
Theoretically if you call the init and the token role, it will generate the token for your.
The link role uses the in-memory tokens produced by the token role and saved as a host
variable named `generatedToken` under the target host variables.

It sounds valid to me to have a `staticToken` variable under the respective target host
assuming that, eventually, you could use external tools like Vault to pull the token from in
a previous step.